ambry-server/src/main/java/com/github/ambry/server/AmbryServerSecurityServiceFactory.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2020 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package com.github.ambry.server;
+
+import com.github.ambry.clustermap.ClusterMap;
+import com.github.ambry.commons.ServerMetrics;
+import com.github.ambry.config.ServerConfig;
+import com.github.ambry.config.VerifiableProperties;
+
+/**
+ * Default implementation of {@link ServerSecurityServiceFactory} for Ambry
+ * <p/>
+ * Returns a new instance of {@link AmbryServerSecurityService} on {@link #getServerSecurityService()} call.
+ */
+public class AmbryServerSecurityServiceFactory implements ServerSecurityServiceFactory {
+
+  private final ServerConfig serverConfig;
+  private final ServerMetrics serverMetrics;
+
+  public AmbryServerSecurityServiceFactory(VerifiableProperties verifiableProperties, ServerMetrics serverMetrics,
+      ClusterMap clusterMap) {
```

Comment:
Is ClusterMap needed?

Reply:
this is needed in our ambryli repo to get the metricsregistry.

Reply:
Why not just pass MetricRegistry into the factory? That is typically what is done for the other factories in the frontend.

Reply:
I changed the constructor to pass the MetricRegistry directly. Also @cgtz as discussed 
1. I have fixed the intTest failures by recreating the AmbryServer objects (reinitServer/reinitServers) when a server restart was intended in the test workflow. 
2. I could not change the ClusterAgentsFactory interface to include getMetricRegistry() api as few implementations(RecoveryTestClusterAgentsFactory) does not have their own metricRegistry. Also this change is not needed actually since intTest failures are fixed by properly reinitailizing the server objects.